Reducers/BreedDetailReducer.swift

```diff
@@ -0,0 +1,286 @@
+import Foundation
+import ComposableArchitecture
+import Combine
+
+@Reducer
+struct BreedDetailReducer {
+    // MARK: - State
+    @ObservableState
+    struct State: Equatable {
+        // Core
+        var breed: CatBreed
+
+        // Favorites
+        var isFavorite: Bool = false
+        var lastErrorMessage: String?
+
+        // Detail loading state
+        enum ScreenState: Equatable {
+            case loading
+            case content
+            case error(String)
+        }
+        var screenState: ScreenState = .loading
+        var isLoading: Bool = false
```

Comment:
```suggestion
        var isLoading: Bool { screenState == .loading }
```
it's better to be a computed var so you don't need to set to true manually, the source of truth is the state